### PR TITLE
Add source link support

### PR DESCRIPTION
**Pull request by bording**

### CODE REVIEW

**Comment by adamralph** [comment]
This one can be rebased now that #385 is merged.

**Comment by bording** [comment]
This has been rebased on to #396, so that needs to be merged, and then this can be rebased to develop.

Now that Mono.Cecil has been fixed (and the fix in included in Fody 2.1.2) this will now work! :tada:

**Comment by bording** [comment]
This should be ready now as well.

File: src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj [hunk 1]
```diff
@@ -26,6 +26,11 @@
     <PackageReference Include="RabbitMQ.Client" Version="[4.1.0, 4.2.0)" PrivateAssets="None" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="SourceLink.Create.GitHub" Version="2.1.2" PrivateAssets="All" />
+    <DotNetCliToolReference Include="dotnet-sourcelink-git" Version="2.1.2" />
+  </ItemGroup>
+
```
Comment (adamralph) [inline src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj]: What does this do for us? Does it introduce something which gets invoked by the TC config?

File: src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj [hunk 2]
```diff
@@ -26,6 +26,11 @@
     <PackageReference Include="RabbitMQ.Client" Version="[4.1.0, 4.2.0)" PrivateAssets="None" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="SourceLink.Create.GitHub" Version="2.1.2" PrivateAssets="All" />
```
Comment (adamralph) [inline src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj]: I presume this adds the necessary build targets which auto-magically inject themselves into the build to do everything that's required?

File: src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj [hunk 3]
```diff
@@ -26,6 +26,11 @@
...
+    <DotNetCliToolReference Include="dotnet-sourcelink-git" Version="2.1.2" />
```
Comment (adamralph) [inline src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj]: Do we need this?

File: src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj (hunk 2 above)
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj]: Exactly.

File: src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj (hunk 3 above)
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ/NServiceBus.Transport.RabbitMQ.csproj]: Yes, it's used by the targets introduced by `SourceLink.Create.GitHub` package.

**Pull request merged**